scripts/release.sh

```diff
@@ -261,8 +261,15 @@ main() {
     for TARGET_ARCH in "amd64" "arm64" "ppc64le" "s390x"; do
       log_callout "Pushing container images to quay.io ${RELEASE_VERSION}-${TARGET_ARCH}"
       maybe_run docker push "quay.io/coreos/etcd:${RELEASE_VERSION}-${TARGET_ARCH}"
+      maybe_run docker manifest create "quay.io/coreos/etcd:${RELEASE_VERSION}-${TARGET_ARCH}" "quay.io/coreos/etcd:${RELEASE_VERSION}-${TARGET_ARCH}"
```

Comment:
Please integrate it with existing block for creating manifest list on line 275.

Reply:
> Please integrate it with existing block for creating manifest list on line 275.

Done, PTAL

Reply:
@serathius Why do you think the new place is better ? 
I actually liked the previous version more: 

The loop in 261 is responsible for preparing the arch-specific images and pushing them. 

The loop in 275 is preparing a single multi-arch image. 

After the modification the code got mixed and in my opinion is more confusing.

Reply:
I was thinking that separating image vs manifest operation will make it cleaner. However I'm not sure after seeing it. 
Delegating to @ptabor and @justinsb to review.